gravitee-rest-api-management/gravitee-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/ApiSubscriptionsResource.java

```diff
@@ -131,15 +137,28 @@ public Response createSubscription(
             @ApiParam(name = "application", required = true)
             @NotNull @QueryParam("application") String application,
             @ApiParam(name = "plan", required = true)
-            @NotNull @QueryParam("plan") String plan) {
+            @NotNull @QueryParam("plan") String plan,
+            @ApiParam(name = "customApiKey")
+            @CustomApiKey @QueryParam("customApiKey") String customApiKey) {
+
+        if (StringUtils.isNotEmpty(customApiKey)  &&
+                !parameterService.findAsBoolean(Key.PLAN_SECURITY_APIKEY_CUSTOM_ALLOWED)) {
+            return Response.status(Response.Status.BAD_REQUEST)
```

Comment:
Add information in `@ApiResponses` that the service can return a BAD_REQUEST

Reply:
resolved